src/Compilers/CSharp/Portable/Symbols/Source/ModifierUtils.cs

```diff
@@ -63,26 +63,64 @@ internal static class ModifierUtils
         }
 
         internal static void ReportDefaultInterfaceImplementationModifiers(
-            LanguageVersion availableVersion,
-            LanguageVersion requiredVersion,
+            bool hasBody,
             DeclarationModifiers modifiers,
             DeclarationModifiers defaultInterfaceImplementationModifiers,
             Location errorLocation,
             DiagnosticBag diagnostics)
         {
-            DeclarationModifiers errorModifiers = modifiers & defaultInterfaceImplementationModifiers;
-            var requiredVersionArgument = new CSharpRequiredLanguageVersion(requiredVersion);
-            var availableVersionArgument = availableVersion.ToDisplayString();
-            while (errorModifiers != DeclarationModifiers.None)
+            if (!hasBody && (modifiers & defaultInterfaceImplementationModifiers) != 0)
             {
-                DeclarationModifiers oneError = errorModifiers & ~(errorModifiers - 1);
-                Debug.Assert(oneError != DeclarationModifiers.None);
-                errorModifiers = errorModifiers & ~oneError;
-                diagnostics.Add(ErrorCode.ERR_DefaultInterfaceImplementationModifier, errorLocation,
-                                ConvertSingleModifierToSyntaxText(oneError),
-                                availableVersionArgument,
-                                requiredVersionArgument);
+                LanguageVersion availableVersion = ((CSharpParseOptions)errorLocation.SourceTree.Options).LanguageVersion;
+                LanguageVersion requiredVersion = MessageID.IDS_DefaultInterfaceImplementation.RequiredVersion();
+                if (availableVersion < requiredVersion)
```

Comment:
I prefer to avoid hard-coding the target version. Also, the version values are used for error reporting below.